File: apps/legacy_vault/legacy_vault/Core/Models/VoiceLogEntry.swift
import Foundation
import CoreData

// MARK: - VoiceLogEntry

struct VoiceLogEntry: Identifiable, Codable, Hashable {
    let id: UUID
    var title: String
    var recordingDate: Date
    var transcript: String
    var aiSummary: String?
    var aiEnrichment: String?
    var sentiment: Int
    var durationMs: Int
    var audioURL: URL?
    var keywords: [String]

    var keywordsJSON: String {
        do {
            return try JSONEncoder().encode(keywords).iso8859_escaped()
        } catch {
            return "[]"
        }
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: VoiceLogEntry, rhs: VoiceLogEntry) -> Bool {
        lhs.id == rhs.id
    }
}

// MARK: - Helpers

private extension Data {
        func iso8859_bytes() -> [UInt8] {
        var result: [UInt8] = []
        self.withUnsafeBytes { rawBuffer in
            let bytes = rawBuffer.bindMemory(to: UInt8.self).baseAddress!
            for index in 0 ..< self.count {
                let byte = bytes[index]
                if byte < 128 {
                    result.append(byte)
                } else {
                    result.append(0x3F)
                }
            }
        }
        return result
    }
}

private extension Data {
        func iso8859_escaped() -> String {
        let bytes = iso8859_bytes()
        return String(bytes: bytes, encoding: .ascii) ?? ""
    }
}

File: apps/legacy_vault_android/app/src/main/java/com/sfx/legacyvault/core/models/VoiceLogEntry.kt
package com.sfx.legacyvault.core.models

import java.util.UUID

data class VoiceLogEntry(
    val id: String = UUID.randomUUID().toString(),
    val title: String = "",
    val recordingDate: Long = 0L,
    val transcript: String = "",
    val aiSummary: String? = null,
    val aiEnrichment: String? = null,
    val sentiment: Int = 0,
    val durationMs: Int = 0,
    val audioURL: String? = null,
    val keywords: List<String> = emptyList(),
    val embeddingId: String? = null
) {
    val keywordsJSON: String
        get() {
            return try {
                keywords.joinToString(",")
            } catch (e: Exception) {
                ""
            }
        }
}
